Jenkinsfile

```diff
@@ -13,11 +13,39 @@ void buildOnYoctoNode(String variant, String image) {
 
         checkout scm
         def manifests = load "ci-scripts/yocto.groovy"
+        try {
+            manifests.buildManifest(variant, image)
+        } catch (e) {
+            // This part will be run in case of fail
+            manifest.shutdownVagrant()
+            throw e
+        } finally {
+            // Can be redundant, so just to doublecheck
+            manifest.shutdownVagrant()
```

Comment:
I do not think this will call shutdownVagrant() (or the extra added one above) in more cases when build is aborted than before. There is a `finally` block with shutdownVagrant() already in buildManifest(). (It will however call it more than necessary for sure if build is not aborted.)

Reply:
I've checked it on my personal repo and it worked
https://pelux.io/jenkins/job/dm_testing/job/master/63/console

Reply:
I was under the impression that Jenkins sent `SIGINT` (or `SIGTERM?) to the command started with `sh` and then just aborted the script when user pressed abort again. I have not tested this though. @tahatrue did most of the experimenting here.

If the script actually continues even when forcefully quitting (user presses "really stop now" or whatever it says) and sends `SIGINT` (or some other signal) we probably want to put this closer to the `sh` call that actually does `vagrant shutdown`. Perhaps even add a loop in `shutdownVagrant()` that that tries until it succeeds or a limit has been reached.

Reply:
In fact, there is already a `retry(5)` in shutdownVagrant(). Does that not work? Is a try/catch/finally still needed inside of it?

Would be nice to find out how it actually works from some Jenkins documentation and how to make sure that a `sh` really should not be aborted.

I found https://gist.github.com/datagrok/dfe9604cb907523f4a2f after a quick search. `sh` and `retry` documentation should be somewhere in https://jenkins.io/doc/book/pipeline/syntax/ ? Can not find it after a quick glance. :/